src/main/java/teammates/ui/output/FeedbackSessionData.java

```diff
@@ -89,12 +95,12 @@ public FeedbackSessionData(FeedbackSessionAttributes feedbackSessionAttributes)
         if (feedbackSessionAttributes.isOpened()) {
             this.submissionStatus = FeedbackSessionSubmissionStatus.OPEN;
         }
-        if (feedbackSessionAttributes.isClosed()) {
-            this.submissionStatus = FeedbackSessionSubmissionStatus.CLOSED;
-        }
         if (feedbackSessionAttributes.isInGracePeriod()) {
             this.submissionStatus = FeedbackSessionSubmissionStatus.GRACE_PERIOD;
         }
+        if (feedbackSessionAttributes.isClosed()) {
+            this.submissionStatus = FeedbackSessionSubmissionStatus.CLOSED;
+        }
```

Comment:
I realise now it looks like I am merely repositioning, which was unintended. I previously rewrote that whole chunk with `isXXXforInstuctor()` in [this earlier commit](https://github.com/TEAMMATES/teammates/pull/11606/commits/99b15073ca88033a5c9f10f81f5b8203c81a87d6), and since I rewrote all of them, I did it in an ordering that I chose. However, I changed them back in [this later commit](https://github.com/TEAMMATES/teammates/pull/11606/commits/b4530946bfe014098775a5fc3613b230621f19f6), but I did not change it back to the original ordering, which makes it look like I merely repositioned it (it looks quite amusing now that you point it out).

Still, I did have a rationale for having that ordering when I did the earlier commit. I was thinking of the probably very rare case when a session is within the grace period (so not open) yet very near to the "closed" time but not quite there yet. In that case, it checks `isClosed` first, and it returns false so nothing is assigned to `submissionStatus` again. After that check, the time finally goes past the "closed" time, right before it checks `isInGracePeriod`, where it then returns false again because it is now actually closed, and nothing is assigned to `submissionStatus` again. If the grace period were long, the other state checks before most likely would have also returned false and nothing is ever assigned to `submissionStatus`. To prevent this, I did it in that order.

I think that since it still applies, maybe I shouldn't move the ordering back?

Reply:
Thanks for helping me understand the rationale.

If you are talking about race condition, it happens all the time with the use of Datastore which is eventually consistent.
We cannot guarantee the data to be consistent all the time, and hence such cases are rare but inevitable elsewhere.

If changing the order mitigate this and there is no other side effect, I'm good with it.